app/src/components/market/market_buy.tsx

```diff
@@ -247,13 +256,13 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
                 }
                 placeholderText={collateral.symbol}
               />
-              <ToggleTokenLock amount={amount} collateral={collateral} context={context} />
             </>
           }
           note={noteAmount}
           title={'Total cost'}
           tooltip={{ id: 'amount', description: 'Shares to buy with this amount of collateral.' }}
         />
+        {showSetAllowance && <button onClick={unlockCollateral}>Set allowance{allowance === null && '...'}</button>}
```

Comment:
When allowance is null the button could be disabled

Reply:
Fixed, thanks.